### PR TITLE
fix identify Azureus style without trailing

### PR DESCRIPTION
**Pull request by SeaHOH**
There are some mistakes in #6098, fixed here.

### CODE REVIEW

File: src/identify_client.cpp [hunk 1]
```diff
@@ -71,7 +71,9 @@ namespace {
 		if (id[0] != '-' || !is_print(char(id[1])) || (id[2] < '0')
 			|| (id[3] < '0') || (id[4] < '0')
 			|| (id[5] < '0') || (id[6] < '0')
-			|| id[7] != '-')
+			|| (id[7] != '-'
+				&& !std::equal(id.begin() + 1, id.begin() + 3, "FG")
+				&& !std::equal(id.begin() + 1, id.begin() + 3, "SP")))
```
Comment (arvidn) [inline src/identify_client.cpp:76]: it doesn't quite seem appropriate to have this exception here. If people are sloppy with how they encode the Azureus-style, perhaps this function should just be more forgiving about it in general

**Comment by arvidn** [comment]
I think this needs to be rebased on top of `RC_2_0` as well

File: src/identify_client.cpp (hunk 1 above)
Comment (SeaHOH) [reply src/identify_client.cpp:76]: Ok, "SP" should be moved, not "FG", I will fix it later.

Comment (arvidn) [reply src/identify_client.cpp:76]: why does "FG" belong here?

Comment (SeaHOH) [reply src/identify_client.cpp]: I am not good at English, it means this commit, not the next fix.

**Comment by SeaHOH** [comment]
It has been rebased now.

**Review by arvidn — APPROVED** [review]

**Pull request merged**